pkg/addonmanager/controllers/agentdeploy/utils.go

```diff
@@ -216,63 +223,100 @@ func (m *managedManifest) deployable(hostedModeEnabled bool, installMode string,
 	return false, nil
 }
 
-func (m *managedManifest) manifestWorkName(addonNamespace, addonName string) string {
-	return constants.DeployWorkName(addonName)
+func (m *managedManifest) manifestWorkNamePrefix(addonNamespace, addonName string) string {
+	return constants.DeployWorkNamePrefix(addonName)
 }
 
 func (m *managedManifest) preDeleteHookManifestWorkName(addonNamespace, addonName string) string {
 	return constants.PreDeleteHookWorkName(addonName)
 }
 
-// buildManifestWorkFromObject returns the deploy manifestwork and preDelete manifestwork, if there is no manifest need
+// BuildDeployWorks returns the deploy manifestWorks. if there is no manifest need
 // to deploy, will return nil.
-func (b *manifestWorkBuiler) buildManifestWorkFromObject(
-	manifestWorkNamespace string,
+func (b *addonWorksBuilder) BuildDeployWorks(addonWorkNamespace string,
 	addon *addonapiv1alpha1.ManagedClusterAddOn,
-	objects []runtime.Object) (deployManifestWork, hookManifestWork *workapiv1.ManifestWork, err error) {
-	var deployManifests []workapiv1.Manifest
-	var hookManifests []workapiv1.Manifest
-	var manifestConfigs []workapiv1.ManifestConfigOption
-
-	addonName := addon.Name
+	existingWorks []workapiv1.ManifestWork,
+	objects []runtime.Object,
+	manifestOptions []workapiv1.ManifestConfigOption) (deployWorks, deleteWorks []*workapiv1.ManifestWork, err error) {
+	var deployObjects []runtime.Object
+	var owner *metav1.OwnerReference
 	installMode, _ := constants.GetHostedModeInfo(addon.GetAnnotations())
+	if installMode == constants.InstallModeDefault {
```

Comment:
add a comment here, hosted mode should not have ownerref.

Reply:
added